src/components/NcDialog/NcDialog.vue

```diff
@@ -356,6 +446,10 @@ export default defineComponent({
 		 * Handle clicking a dialog button -> should close
 		 */
 		const handleButtonClose = () => {
+			// Skip close if invalid dialog
+			if (dialogTagName.value === 'form' && !dialogElement.value.reportValidity()) {
+				return
+			}
 			handleClosing()
 			window.setTimeout(() => handleClosed(), 300)
 		}
```

Comment:
Note for future - it's more common case when a dialog button should not close it. This behavior forces to use custom buttons instead of dialog buttons.

Reply:
Maybe allow returning `false` from the callback to prevent close?